### PR TITLE
Merge QR Code Login into Trunk

### LINKED ISSUE #18594: Project: iOS QRCode Login Support
## Description
Allow WP.com users to sign in to their browsers by scanning a QR Code in the WPMobile iOS App.

## Tasks:
- [x] Add a way to access the QRCode scanner via the Me tab: https://github.com/wordpress-mobile/WordPress-iOS/pull/18715

### Create a view controller for QR Code Scanning
- [x] Add camera capture: https://github.com/wordpress-mobile/WordPress-iOS/pull/18716
- [x] Add a square overlay to the camera: https://github.com/wordpress-mobile/WordPress-iOS/pull/18716
- [x] Add basic link validation: https://github.com/wordpress-mobile/WordPress-iOS/pull/18716
- [x] Add handling when a valid link is scanned: https://github.com/wordpress-mobile/WordPress-iOS/pull/18716
 
#### Design a basic view
- [x] Add a header label: https://github.com/wordpress-mobile/WordPress-iOS/pull/18752
- [x] Add a label that displays the location of the browser: https://github.com/wordpress-mobile/WordPress-iOS/pull/18756
- [x] Add a label that displays the vendor of the browser: https://github.com/wordpress-mobile/WordPress-iOS/pull/18756
- [x] Add a button to "Accept": https://github.com/wordpress-mobile/WordPress-iOS/pull/18752
- [x] Add a button to "Reject": https://github.com/wordpress-mobile/WordPress-iOS/pull/18752
#### Validate the incoming link with the server
- [x] Display a loading indicator: https://github.com/wordpress-mobile/WordPress-iOS/pull/18752
- [x] Make call to validate endpoint: https://github.com/wordpress-mobile/WordPress-iOS/pull/18756
- [x] Push to the error view if request fails: https://github.com/wordpress-mobile/WordPress-iOS/pull/18756
- [x] Display main UI when request succeeds: https://github.com/wordpress-mobile/WordPress-iOS/pull/18752
#### Add button actions
- [x] Dismiss the "flow" when the "reject" button is tapped: https://github.com/wordpress-mobile/WordPress-iOS/pull/18752
- [x] Add "Accept" button action: https://github.com/wordpress-mobile/WordPress-iOS/pull/18752
- [x] Make call to the authorize endpoint: https://github.com/wordpress-mobile/WordPress-iOS/pull/18756
- [x] Display another loading UI: https://github.com/wordpress-mobile/WordPress-iOS/pull/18752
- [x] Push to the error view if the request fails: https://github.com/wordpress-mobile/WordPress-iOS/pull/18756
- [x] Push to the "Authorized" view on success: https://github.com/wordpress-mobile/WordPress-iOS/pull/18752
 
#### Design a basic view
- [x] Add a header label: https://github.com/wordpress-mobile/WordPress-iOS/pull/18752
- [x] Add a label that directs the user to go back to the browser: https://github.com/wordpress-mobile/WordPress-iOS/pull/18752
- [x] Add a button to "Dismiss": https://github.com/wordpress-mobile/WordPress-iOS/pull/18752
 
#### Create the new QRLoginRemoteService in WPKit
  - [x] Add a validate link method: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/513
      - [x] Create a new Decodable struct for the success response
      - [x] Map success to new struct
iOS/pull/513
  - [x] Add an authorize method: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/513
      - [x] Create a new Decodable struct for the success response
      - [x] Map success to new struct
  - [x] Create a WPMobile QRLoginService: https://github.com/wordpress-mobile/WordPress-iOS/pull/18756
      - [x] Add a validate link method
      - [x] Add a authorize method

### Update Deep Linking
- [x] Add handling for the QRCode URL when scanned outside of the app: https://github.com/wordpress-mobile/WordPress-iOS/pull/18765

### Add Tracking
- [x] Add tracking to the QRCode scanning view: https://github.com/wordpress-mobile/WordPress-iOS/pull/18766
- [x] Add tracking to the Are you sure view: https://github.com/wordpress-mobile/WordPress-iOS/pull/18766
- [x] Add tracking to the Authorized view: https://github.com/wordpress-mobile/WordPress-iOS/pull/18766
- [x] Add tracking to the deep link handler: https://github.com/wordpress-mobile/WordPress-iOS/pull/18766

### Finishing Touches
- [x] Add unit tests: https://github.com/wordpress-mobile/WordPress-iOS/pull/18924
- [x] Fix iPad layout

### PR DESCRIPTION
**Pull request by emilylaguna**
Fixes #18594

Merges the QR Login project into trunk, and makes the final changes to release the feature:
- Makes it Jetpack only
- Enables the feature flag for all
- Updates WPKit to beta

### To test:
- Make sure the tests are 🟢 
- Verify the release notes are 👍 

## Regression Notes
1. Potential unintended areas of impact
Existing me menu items, deep link handling

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
None

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by emilylaguna** [comment]
@momo-ozawa @hassaanelgarem if https://github.com/wordpress-mobile/WordPress-iOS/pull/18924 is approved can you ✅  give this a review as well so I can merge this feature into trunk before the freeze? 🙇‍♀️

**Comment by emilylaguna** [comment]
I'll take care of any conflicts after #18924 is merged in

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 20.2
 -----
-
+* [***] [Jetpack-only] Enables QR Code Login scanning from the Me menu. [#18904]
```
Comment (momo-ozawa) [inline RELEASE-NOTES.txt:3]: 👍

**Review by momo-ozawa — APPROVED** [review]
🚀

**Pull request merged**